java/semantickernel-core/src/main/java/com/microsoft/semantickernel/orchestration/NativeSKFunction.java

```diff
@@ -346,7 +357,54 @@ private static String getArgumentValue(
                         "Unknown arg " + parameter.getName());
             }
         }
-        return arg;
+
+        SKFunctionParameters annotation = parameter.getAnnotation(SKFunctionParameters.class);
+        if (annotation == null || annotation.type() == null) {
+            return arg;
+        }
+        Class<?> type = annotation.type();
+        if (Number.class.isAssignableFrom(type)) {
+            arg = arg.replace(",", ".");
+        }
+
+        Object value = arg;
+        // Well-known types only
+        Function converter = converters.get(type);
+        if (converter != null) {
+            try {
+                value = converter.apply(arg);
+            } catch (NumberFormatException nfe) {
+                throw new AIException(
+                        AIException.ErrorCodes.InvalidConfiguration,
+                        "Invalid value for "
+                                + parameter.getName()
+                                + " expected "
+                                + type.getSimpleName()
+                                + " but got "
+                                + arg);
+            }
+        }
+        return value;
+    }
+
+    private static final Map<Class<?>, Function<String, ?>> converters = new HashMap<>();
```

Comment:
@johnoliver You had a question about these converters having Boolean and boolean (for example). The reason for this is that the user might put `@SKFunctionParameters(type=boolean.class)`.  The other option would be to box the return value from the annotation's `type()` method, but you'd still end up with some sort of `if (type==boolean.class) type = Boolean.class` logic.

Another question you had was whether we could add Iterable to the converters and get rid of this bit of code
```
                            if (it instanceof Iterable) {
                                // Handle return from things like Mono<List<?>>
                                // from {{function 'input'}} as part of the prompt.
                                it = ((Iterable<?>) it).iterator().next();
                            }
```
The converters are for converting a parameter value according to the `type` in the SKFunctionParameters annotation. I don't think we allow Iterable<?> as a parameter to a native function, do we?